pkg/validate/validate.go

```diff
@@ -115,6 +115,17 @@ func ClusterName1035(v string) error {
 	return ClusterName(v)
 }
 
+// GCPClusterName checks if the provided cluster name has words similar to the word 'google'
+// since resources with that name are not allowed in GCP.
+func GCPClusterName(v string) error {
+	reStartsWith := regexp.MustCompile("^goog")
+	reContains := regexp.MustCompile(".*g[o0]{2}gle.*")
+	if reStartsWith.MatchString(v) || reContains.MatchString(v) {
+		return errors.New("names that start with 'goog' or contain google are reserved in GCP and are not allowed")
```

Comment:
There is inconsistent quoting. The `goog` is quoted with single quotes and the `google` has no quoting. I recommend quoting both with double quotes. Let's also remove the text around why we are rejecting the name.
```suggestion
		return errors.New("cluster name must not start with \"goog\" or contain variations of \"google\"")
```